lsp-mode.el

```diff
@@ -7979,6 +7979,20 @@ Errors if there are none."
   (lsp--warn "Stopping %s" (lsp--workspace-print workspace))
   (with-lsp-workspace workspace (lsp--shutdown-workspace)))
 
+(defun lsp-workspace-kill (workspace &optional restart)
+  "Kill the language server behind WORKSPACE.
+Prefer `lsp-workspace-shutdown' if possible, since this command
+may leave the server and Emacs in an inconsistent state, so use
+only in an emergency.
+
+If RESTART (prefix-arg) is non-nil, cause the server to be
+restarted afterwards."
+  (interactive (list (lsp--read-workspace) current-prefix-arg))
+  (lsp--warn "Killing %s" (lsp--workspace-print workspace))
+  (setf (lsp--workspace-shutdown-action workspace)
+        (if restart 'restart 'shutdown))
+  (kill-process (lsp--workspace-proc workspace)))
```

Comment:
This needs to be `lsp--workspace-cmd-proc` for the server process to be killed, else you will just kill the connection in the case of tcp server.
The `lsp-workspace-shutdown/restart` is already killing (and restarting) the server process already. Can you elaborate more on why we need this?

Reply:
`lsp-workspace-shutdown` is more graceful, so won't work if a server hangs. `lsp-workspace-kill` just kills the server, which will bring it down immediately most of the time (and as such should only be used for debugging).

Reply:
> `lsp-workspace-shutdown` is more graceful, so won't work if a server hangs.

IMHO we should make sure that lsp-workspace-shutdown works even with a hanging server.

Reply:
Looking at `lsp--shutdown-workspace`, if the server hangs, after the time out we will call `lsp--uninitialize-workspace`, which kills the server process too.

Reply:
Actually, it won't, due to `error` exiting the function on timeout:
```
(let ((lsp-response-timeout 0.01))
  (condition-case nil (with-current-buffer "<some lsp buffer>" (lsp-request "textDocument/documentSymbol" '()))
    (error t)))
 ```
 
 Perhaps `lsp-workspace-shutdown` should be async? I don't really have much justification for `lsp-workspace-kill` other than "it might be useful as a debugging tool sometime", so feel free to close this PR.